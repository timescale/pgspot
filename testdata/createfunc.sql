
-- should warn about unsafe function and format
CREATE OR REPLACE FUNCTION unsafe3() RETURNS TEXT LANGUAGE SQL AS $$ SELECT unsafe_call3('%I','foo'); $$;
-- should warn about call
CREATE FUNCTION safe5() RETURNS TEXT LANGUAGE SQL AS $$ SELECT unsafe_call5('%I','foo'); $$;

-- safe
CREATE FUNCTION f8() RETURNS TEXT LANGUAGE SQL AS $$ SELECT unsafe_call8('%I','foo'); $$;

-- should not warn about since f2 was created in this script so replace is safe
CREATE OR REPLACE FUNCTION f8() RETURNS TEXT LANGUAGE SQL AS $$ SELECT pg_catalog.safe_call11('%I','foo'); $$;

-- not safe since signature is different
CREATE OR REPLACE FUNCTION f8(int) RETURNS TEXT LANGUAGE SQL AS $$ SELECT pg_catalog.safe_call14('%I','foo'); $$;

CREATE FUNCTION safe16() RETURNS TEXT LANGUAGE plpgsql AS $$
BEGIN
  RETURN unsafe_call18('%I','foo');
END; $$;

-- safe creation
CREATE FUNCTION f22(a int = 1) RETURNS TEXT LANGUAGE SQL AS $$ SELECT 'foo'; $$ SET search_path TO pg_catalog,pg_temp;

-- safe as well since it was previously created with same signature but different default
CREATE OR REPLACE FUNCTION f22(a int = 2) RETURNS TEXT LANGUAGE SQL AS $$ SELECT 'foo'; $$ SET search_path TO pg_catalog,pg_temp;

-- not safe since it's new signature
CREATE OR REPLACE FUNCTION f22(b int) RETURNS TEXT LANGUAGE SQL AS $$ SELECT 'foo'; $$ SET search_path TO pg_catalog,pg_temp;
