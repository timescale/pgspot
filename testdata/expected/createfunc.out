PS002: Unsafe function creation: unsafe3() at line 2
PS005: Function without explicit search_path: unsafe3() at line 2
PS016: Unqualified function call: unsafe_call3 at line 2
PS005: Function without explicit search_path: safe5() at line 4
PS016: Unqualified function call: unsafe_call5 at line 4
PS005: Function without explicit search_path: f8() at line 6
PS016: Unqualified function call: unsafe_call8 at line 6
PS005: Function without explicit search_path: f8() at line 9
PS002: Unsafe function creation: f8(integer) at line 12
PS005: Function without explicit search_path: f8(integer) at line 12
PS005: Function without explicit search_path: safe16() at line 15
PS016: Unqualified function call: unsafe_call18 at line 15
PS002: Unsafe function creation: f22(b integer) at line 26

 Errors: 3 Warnings: 10 Unknown: 0 

